zavod/zavod/context.py

```diff
@@ -316,19 +372,61 @@ def fetch_html(
             headers=headers,
             auth=auth,
             cache_days=cache_days,
+            method=method,
+            data=data,
         )
         if text is not None and len(text):
             try:
                 return html.fromstring(text)
             except Exception:
-                self.clear_url(url, params)
+                fingerprint = self.get_request_fingerprint(
+                    url=url, auth=auth, method=method, data=data
+                )
+                self.clear_url(fingerprint)
                 raise
         raise ValueError("Invalid HTML document: %s" % url)
 
-    def clear_url(self, url: str, params: ParamsType = None) -> None:
-        """Remove a given URL from the cache."""
-        url = build_url(url, params)
-        self.cache.delete(url)
+    def clear_url(self, fingerprint: str) -> None:
+        """
+        Remove a given URL from the cache using request fingerprint
+        Args:
+            fingerprint: The unique fingerprint of the request.
+        Returns:
+            None
+        """
+        self.cache.delete(fingerprint)
+
+    def get_request_fingerprint(
+        self,
+        url: str,
+        auth: Optional[_Auth] = None,
+        method: str = "GET",
+        data: Any = None,
+    ) -> str:
+        """
+        Generate a unique fingerprint for an HTTP request.
+        Args:
+            url: The URL of the request.
+            auth: The authentication credentials.
+            method: The HTTP method of the request.
+            data: The data to be sent in the request body.
+        Returns:
+            A unique fingerprint for the request (url + hashed payload).
+        """
+
+        hsh = hashlib.sha1(
```

Comment:
Can we use `from banal import hash_data` for this? It's used in other places of the framework and will handle things like sets.

Reply:
Sure. Let me know if you want to switch orjson.loads back to json.loads too.

Reply:
The thing coming out of `hash_data` is already a sha1 :)

Reply:
I mean this one: https://github.com/opensanctions/opensanctions/pull/717/files#diff-087e34be4ac7c0656253b06a5737f1de4d1abab35f78df0f286828c60f618b9cL286

Reply:
Ah no that makes sense!